policyengine_api/routes/household_routes.py

```diff
@@ -0,0 +1,137 @@
+)
+
+
+household_bp = Blueprint("household", __name__)
+household_service = HouseholdService()
+
+
+@household_bp.route(
+    "/<country_id>/household/<int:household_id>", methods=["GET"]
+)
+@validate_country
+def get_household(country_id: str, household_id: int) -> Response:
+    """
+    Get a household's input data with a given ID.
+
+    Args:
+        country_id (str): The country ID.
+        household_id (int): The household ID.
+    """
+    print(f"Got request for household {household_id} in country {country_id}")
+
+    household: dict | None = household_service.get_household(
+        country_id, household_id
+    )
+    if household is None:
+        raise NotFound(f"Household #{household_id} not found.")
+    else:
+        return Response(
+            json.dumps(
+                {
+                    "status": "ok",
+                    "message": None,
+                    "result": household,
+                }
+            ),
+            status=200,
+            mimetype="application/json",
+        )
+
+
+@household_bp.route("/<country_id>/household", methods=["POST"])
+@validate_country
+def post_household(country_id: str) -> Response:
+    """
+    Set a household's input data.
+
+    Args:
+        country_id (str): The country ID.
+    """
+
+    # Validate payload
+    payload = request.json
+    is_payload_valid, message = validate_household_payload(payload)
+    if not is_payload_valid:
+        raise BadRequest(f"Unable to create new household; details: {message}")
+
+    # The household label appears to be unimplemented at this time,
+    # thus it should always be 'None'
+    label: str | None = payload.get("label")
+    household_json: dict = payload.get("data")
+
+    household_id = household_service.create_household(
+        country_id, household_json, label
+    )
+
+    return Response(
+        json.dumps(
+            {
+                "status": "ok",
+                "message": None,
+                "result": {
+                    "household_id": household_id,
+                },
+            }
+        ),
+        status=201,
+        mimetype="application/json",
+    )
+
+
+@household_bp.route(
+    "/<country_id>/household/<int:household_id>", methods=["PUT"]
+)
+@validate_country
+def update_household(country_id: str, household_id: int) -> Response:
+    """
+    Update a household's input data.
+
+    Args:
+        country_id (str): The country ID.
+        household_id (int): The household ID.
+    """
+
+    # Validate payload
+    payload = request.json
+    is_payload_valid, message = validate_household_payload(payload)
+    if not is_payload_valid:
+        raise BadRequest(
+            f"Unable to update household #{household_id}; details: {message}"
+        )
+
+    # First, attempt to fetch the existing household
+    label: str | None = payload.get("label")
+    household_json: dict = payload.get("data")
+
+    household: dict | None = household_service.get_household(
```

Comment:
**accoldate** I may have already said this somewhere, but I like explicit types rather than inferred in these circumstances (where the type returned from get_household is not obvious unless you go look at the method).

I don't know I'm on the winning side of that argument in the industry though.

Reply:
Thank you, I've just found it so much easier while building to write these types, especially as it has helped me to better clarify functions of the code while programming it; if the type hinting becomes too convoluted, it's time to rethink.